modules/querier/http.go

```diff
@@ -94,7 +97,7 @@ func (q *Querier) TraceByIDHandler(w http.ResponseWriter, r *http.Request) {
 
 	span.SetTag("response marshalling format", util.JSONTypeHeaderValue)
 	marshaller := &jsonpb.Marshaler{}
-	err = marshaller.Marshal(w, resp.Trace)
+	err = marshaller.Marshal(w, resp)
```

Comment:
There is a difference between our responses depending on whether the caller accept protobuf or not: if the caller accepts protobuf we return `HTTP 206` and `resp.Trace` (line 82).

But if the caller requests something else (aka JSON) we marshal the entire `TraceByIDResponse`. Which would be something like:
```json
{
  "trace": ...,
  "blockErrCount": ...
}
```
Why not also return `HTTP 206`? This change breaks our API I think.